th2_check2_recon/recon.py

```diff
@@ -64,27 +60,34 @@ def stop(self):
             self.__message_router.unsubscribe_all()
             for rule in self.rules:
                 rule.stop()
-            if self.__message_comparator is not None:
-                self.__message_comparator.stop()
-            self.__event_store.stop()
+            if self.message_comparator is not None:
+                self.message_comparator.stop()
+            self.event_store.stop()
         except Exception:
             logger.exception('Error while stop Recon')
         finally:
             self.__loop.close()
             logger.info('Recon stopped!')
 
-    def __load_rules(self, event_store: EventStore) -> [Rule]:
+    def __load_rules(self):
         logger.info('Try load rules')
         rules_package = importlib.import_module(self.__config.rules_package_path)
         loaded_rules = []
         for rule_config in self.__config.rules:
             if rule_config.enabled:
                 module = importlib.import_module(rules_package.__name__ + '.' + rule_config.name)
                 match_timeout = rule_config.match_timeout * 1_000_000_000 + rule_config.match_timeout_offset_ns
-                loaded_rules.append(module.Rule(event_store=event_store,
-                                                message_comparator=self.__message_comparator,
+                loaded_rules.append(module.Rule(recon=self,
                                                 cache_size=self.__config.cache_size,
                                                 match_timeout=match_timeout,
                                                 configuration=rule_config.configuration))
         logger.info('Rules loaded')
         return loaded_rules
+
+    def put_shared_message(self, shared_group_id: str, message: ReconMessage, attributes: tuple):
+        new_message = ReconMessage(message.proto_message)
+        new_message.group_id = shared_group_id
+        for rule in self.rules:
+            groups = rule.description_of_groups()
+            if shared_group_id in groups.keys() and MessageGroupType.shared in groups[shared_group_id]:
```

Comment:
I think the usage of  '==' is more correct in the second comparison.

if shared_group_id in groups.keys() and groups[shared_group_id] == MessageGroupType.shared: